packages/kbn-observable/src/observable.ts

```diff
@@ -0,0 +1,422 @@
+   * to listen for values that an Observable emits, as well as for when it
+   * completes or errors. You can achieve this in two following ways:
+   *
+   * The first way is creating an object that implements {@link Observer} interface. It should have methods
+   * defined by that interface, but note that it should be just a regular JavaScript object, which you can create
+   * yourself in any way you want (ES6 class, classic function constructor, object literal etc.). In particular do
+   * not attempt to use any RxJS implementation details to create Observers - you don't need them. Remember also
+   * that your object does not have to implement all methods. If you find yourself creating a method that doesn't
+   * do anything, you can simply omit it. Note however, that if `error` method is not provided, all errors will
+   * be left uncaught.
+   *
+   * The second way is to give up on Observer object altogether and simply provide callback functions in place of its methods.
+   * This means you can provide three functions as arguments to `subscribe`, where first function is equivalent
+   * of a `next` method, second of an `error` method and third of a `complete` method. Just as in case of Observer,
+   * if you do not need to listen for something, you can omit a function, preferably by passing `undefined` or `null`,
+   * since `subscribe` recognizes these functions by where they were placed in function call. When it comes
+   * to `error` function, just as before, if not provided, errors emitted by an Observable will be thrown.
+   *
+   * Whatever style of calling `subscribe` you use, in both cases it returns a Subscription object.
+   * This object allows you to call `unsubscribe` on it, which in turn will stop work that an Observable does and will clean
+   * up all resources that an Observable used. Note that cancelling a subscription will not call `complete` callback
+   * provided to `subscribe` function, which is reserved for a regular completion signal that comes from an Observable.
+   *
+   * Remember that callbacks provided to `subscribe` are not guaranteed to be called asynchronously.
+   * It is an Observable itself that decides when these functions will be called. For example {@link of}
+   * by default emits all its values synchronously. Always check documentation for how given Observable
+   * will behave when subscribed and if its default behavior can be modified with a {@link Scheduler}.
+   *
+   * @example <caption>Subscribe with an Observer</caption>
+   * const sumObserver = {
+   *   sum: 0,
+   *   next(value) {
+   *     console.log('Adding: ' + value);
+   *     this.sum = this.sum + value;
+   *   },
+   *   error() { // We actually could just remove this method,
+   *   },        // since we do not really care about errors right now.
+   *   complete() {
+   *     console.log('Sum equals: ' + this.sum);
+   *   }
+   * };
+   *
+   * Rx.Observable.of(1, 2, 3) // Synchronously emits 1, 2, 3 and then completes.
```

Comment:
I'll go over some of these and clean up — just forgot when finishing up other parts